lib/elastic_apm/transport/connection/http.rb

```diff
@@ -75,7 +75,15 @@ def close(reason)
           @closed.make_true
 
           @wr&.close
-          return if @request.nil? || @request&.join(5)
+
+          if @request&.join(5)
+            @rd&.close
+            return
+          end
+
+          @rd&.close
+
+          return if @request.nil?
```

Comment:
I would suggest simplifying this to:

```ruby
          @request&.join(5) # will return nil if @request is nil, will return nil if 5s expires, will return the thread if the thread exits
          @rd&.close
          return if @request.nil?
```

because you will always call `@request&.join(5)` and then `@rd&.close` no matter what #join returns.

Reply:
The condition before was `return if @request.nil? || @request&.join(5)`, so we need to return if `@request&.join(5)` returns a "truthy" value. `return if @request.nil?` won't make it return when `@request&.join(5)` returns a truthy value.

Reply:
I understand now, thanks for the explanations.